src/click-outside.directive.ts

```diff
@@ -1,67 +1,89 @@
 import {
-  Directive,
-  ElementRef,
-  EventEmitter,
-  Inject,
-  Input,
-  OnChanges,
-  OnDestroy,
-  OnInit,
-  Output,
-  SimpleChanges
+    Directive,
+    ElementRef,
+    EventEmitter,
+    Inject,
+    Input,
+    OnChanges,
+    OnDestroy,
+    OnInit,
+    Output,
+    SimpleChanges
 } from '@angular/core';
 import { DOCUMENT } from '@angular/platform-browser';
 
 @Directive({ selector: '[clickOutside]' })
 export class ClickOutsideDirective implements OnInit, OnDestroy, OnChanges {
-  @Input() attachOutsideOnClick: boolean = false;
-  @Output() clickOutside: EventEmitter<Event> = new EventEmitter<Event>();
+    @Input() attachOutsideOnClick: boolean = false;
+    @Input() exclude:string = "";
+    private nodeExcluded = [];
+    @Output() clickOutside: EventEmitter<Event> = new EventEmitter<Event>();
 
-  constructor(
-    @Inject(DOCUMENT) private _document /*: HTMLDocument*/,
-    private _el: ElementRef) {
-    this._initOnClickBody = this._initOnClickBody.bind(this);
-    this._onClickBody = this._onClickBody.bind(this);
-  }
+    constructor(
+        @Inject(DOCUMENT) private _document /*: HTMLDocument*/,
+        private _el: ElementRef) {
+        this._initOnClickBody = this._initOnClickBody.bind(this);
+        this._onClickBody = this._onClickBody.bind(this);
+    }
+
+    ngOnInit() {
+        this._init();
+    }
 
-  ngOnInit() {
-    this._init();
-  }
+    ngOnDestroy() {
+        if (this.attachOutsideOnClick) {
+            this._el.nativeElement.removeEventListener('click', this._initOnClickBody);
+        }
+
+        this._document.body.removeEventListener('click', this._onClickBody);
+    }
 
-  ngOnDestroy() {
-    if (this.attachOutsideOnClick) {
-      this._el.nativeElement.removeEventListener('click', this._initOnClickBody);
+    ngOnChanges(changes: SimpleChanges) {
+        if (changes['attachOutsideOnClick'] &&
+            changes['attachOutsideOnClick'].previousValue !== changes['attachOutsideOnClick'].currentValue) {
+            this._init();
+        }
     }
 
-    this._document.body.removeEventListener('click', this._onClickBody);
-  }
+    private _init() {
 
-  ngOnChanges(changes: SimpleChanges) {
-    if (changes['attachOutsideOnClick'] &&
-      changes['attachOutsideOnClick'].previousValue !== changes['attachOutsideOnClick'].currentValue) {
-      this._init();
+        let nodeExcluded = this.exclude.split(',');
```

Comment:
Might be better to call this something else. I was mixing up this with the class's instance of `nodeExcluded` for a bit.